src/mscorlib/shared/System/Globalization/DateTimeParse.cs

```diff
@@ -3607,15 +3607,15 @@ private static bool MatchAbbreviatedTimeMark(ref __DTString str, DateTimeFormatI
             // change the code below.
             if (str.GetNext())
             {
-                if (str.GetChar() == dtfi.AMDesignator[0])
+                if (dtfi.AMDesignator.Length > 0 && str.GetChar() == dtfi.AMDesignator[0])
```

Comment:
Cache AMDesignator in a local?

Reply:
wondering if the JIT will optimize this?

Reply:
It will not. `AMDesignator` is non-trivial property getter.